### PR TITLE
refactor: convert vault-registry pallet to FRAME v2

### PR DESCRIPTION
**Pull request by sander2**

### CODE REVIEW

File: crates/vault-registry/src/lib.rs [hunk 1]
```diff
@@ -69,147 +64,109 @@ pub enum LiquidationTarget {
...
-        /// Mapping of reserved BTC addresses to the registered account
-        ReservedAddresses: map hasher(blake2_128_concat) BtcAddress => T::AccountId;
+        /// The overarching event type.
+        type Event: From<Event<Self>>
+            + Into<<Self as frame_system::Config>::Event>
+            + IsType<<Self as frame_system::Config>::Event>;
 
-        /// Total collateral used for backing tokens issued by active vaults, excluding the liquidation vault
-        TotalUserVaultBackingCollateral: Backing<T>;
+        type RandomnessSource: Randomness<H256, Self::BlockNumber>;
```
Comment (gregdhill) [inline crates/vault-registry/src/lib.rs:96]: Add doc comment here please

File: crates/vault-registry/src/lib.rs [hunk 2]
```diff
@@ -69,147 +64,109 @@ pub enum LiquidationTarget {
...
 
-        /// Build storage at V1 (requires default 0).
-        StorageVersion get(fn storage_version) build(|_| Version::V1): Version = Version::V0;
+        type UnsignedFixedPoint: FixedPointNumber + Encode + EncodeLike + Decode + MaybeSerializeDeserialize;
```
Comment (gregdhill) [inline crates/vault-registry/src/lib.rs:98]: Also on this

File: crates/vault-registry/src/lib.rs [hunk 3]
```diff
@@ -272,49 +237,48 @@ decl_module! {
...
         /// * `VaultNotFound` - if no vault exists for the origin account
         /// * `InsufficientCollateralAvailable` - if the vault does not own enough collateral
-        #[weight = <T as Config>::WeightInfo::withdraw_collateral()]
+        #[pallet::weight(<T as Config>::WeightInfo::withdraw_collateral())]
         #[transactional]
-        fn withdraw_collateral(origin, #[compact] amount: Backing<T>) -> DispatchResult {
+        pub fn withdraw_collateral(
+            origin: OriginFor<T>,
+            #[pallet::compact] amount: Backing<T>,
+        ) -> DispatchResultWithPostInfo {
```
Comment (gregdhill) [inline crates/vault-registry/src/lib.rs:245]: Is `DispatchResultWithPostInfo` now required for all calls?

File: crates/vault-registry/src/lib.rs [hunk 4]
```diff
@@ -325,37 +289,235 @@ decl_module! {
...
 
-        fn on_initialize(_n: T::BlockNumber) -> Weight {
-            match Self::_on_initialize() {
-                Ok(weight) => weight,
-                _ => <T as Config>::WeightInfo::liquidate_undercollateralized_vaults(0)
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    #[pallet::metadata(T::AccountId = "AccountId", Backing<T> = "Backing", Issuing<T> = "Issuing")]
+    pub enum Event<T: Config> {
+        RegisterVault(T::AccountId, Backing<T>),
```
Comment (gregdhill) [inline crates/vault-registry/src/lib.rs:307]: We need more doc comments on these events / errors but not related to this PR

File: crates/vault-registry/src/lib.rs [hunk 5]
```diff
@@ -325,37 +289,235 @@ decl_module! {
...
+        pub premium_redeem_threshold: UnsignedFixedPoint<T>,
+        #[doc = " Determines the lower bound for the collateral rate in issued tokens."]
+        #[doc = " Must be strictly greater than 100000. If a Vault’s collateral rate"]
+        #[doc = " drops below this, automatic liquidation (forced Redeem) is triggered."]
+        pub liquidation_collateral_threshold: UnsignedFixedPoint<T>,
+    }
+
+    #[cfg(feature = "std")]
+    impl<T: Config> Default for GenesisConfig<T>
+    // TODO_MAYBE_WHERE_CLAUSE
```
Comment (gregdhill) [inline crates/vault-registry/src/lib.rs:480]: I think this can be removed

File: crates/vault-registry/src/lib.rs [hunk 6]
```diff
@@ -325,37 +289,235 @@ decl_module! {
...
+    #[pallet::getter(fn storage_version)]
+    pub(super) type StorageVersion<T: Config> = StorageValue<_, Version, ValueQuery, DefaultForStorageVersion>;
+
+    /// Map of Vault Operators
+    #[pallet::storage]
+    pub(super) type IsNominationOperator<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, bool, ValueQuery>;
+
+    #[pallet::genesis_config]
+    pub struct GenesisConfig<T: Config> {
+        #[doc = " The minimum collateral (e.g. DOT/KSM) a Vault needs to provide"]
```
Comment (gregdhill) [inline crates/vault-registry/src/lib.rs:454]: Same as on `btc-relay` conversion, is this structure (`#[doc = ""]`) required or can we just use doc comments?

File: crates/vault-registry/src/lib.rs (hunk 3 above)
Comment (sander2) [reply crates/vault-registry/src/lib.rs:245]: According to the [upgrade guidelines](https://substrate.dev/rustdocs/v3.0.0/frame_support/attr.pallet.html#upgrade-guidelines), yes

File: crates/vault-registry/src/lib.rs (hunk 6 above)
Comment (sander2) [reply crates/vault-registry/src/lib.rs:454]: This was auto-generated, but you are right, regular doc comments work too, I'll change

**Review by gregdhill — APPROVED** [review]

**Pull request merged**